### PR TITLE
handle mruby response array and generator correctly

### PR DESCRIPTION
**Pull request by i110**
Fixes https://github.com/h2o/h2o/issues/1575

### CODE REVIEW

File: lib/handler/mruby/embedded.c.h [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "            while 1\n"                                                                                                        \
     "              resp = app.call(req)\n"                                                                                         \
     "              cached = self_fiber\n"                                                                                          \
-    "              (req, generator) = Fiber.yield(*resp, generator)\n"                                                             \
+    "              (req, generator) = Fiber.yield(resp, generator)\n"                                                              \
```
Comment (kazuho) [inline lib/handler/mruby/embedded.c.h]: Am I correct in assuming that in #1173 we switched from 3-arg return to 4-arg return (i.e. status + headers + body + generator), but forgot to set the 4th argument in some cases?

If that is true, why are we switching to a 3-element array + generator now? Is there a reason?

Comment (i110) [reply lib/handler/mruby/embedded.c.h]: > forgot to set the 4th argument in some cases?

No.  The source of the problem was array expansion operator (`*`). If we expand nil, it produces nothing (i.e. `[*nil]` -> `[]`). In other words, if we expand `resp` here, the position of generator in the array is undeterministic.

Comment (kazuho) [reply lib/handler/mruby/embedded.c.h]: Ah! Thank you for the clarification.

**Comment by kazuho** [comment]
@i110 The code looks good to me. Please merge when you think it is ready.

**Pull request merged**